main.go

```diff
@@ -49,6 +50,7 @@ var (
 	templates                *template.Template
 	analytics                RedisAnalyticsHandler
 	GlobalEventsJSVM         JSVM
+	globalEventsJSVMMu       sync.Mutex
```

Comment:
I also saw a race related to the JSVM, but it was because of the lack of a lock on `apiSpecs`. I'm fairly sure that a lock here isn't necessary. Do you have a race report on master that points directly here?

Reply:
if we discard these changes, i.e.:
```
diff --git a/main.go b/main.go
index 1a88571..35dfff4 100644
--- a/main.go
+++ b/main.go
@@ -183,7 +183,7 @@ func setupGlobals() {
 
        // Set up global JSVM
        if config.Global.EnableJSVM {
-               initGlobalEventsJSVM(nil)
+               GlobalEventsJSVM.Init(nil)
        }
 
        if config.Global.CoProcessOptions.EnableCoProcess {
@@ -662,7 +662,7 @@ func doReload() {
 
        // Reset the JSVM
        if config.Global.EnableJSVM {
-               initGlobalEventsJSVM(nil)
+               GlobalEventsJSVM.Init(nil)
        }
 
        log.WithFields(logrus.Fields{
diff --git a/rpc_backup_handlers.go b/rpc_backup_handlers.go
index 8ebea5e..3b34a1d 100644
--- a/rpc_backup_handlers.go
+++ b/rpc_backup_handlers.go
@@ -92,7 +92,7 @@ func doLoadWithBackup(specs []*APISpec) {
        }
 
        // Reset the JSVM
-       initGlobalEventsJSVM(nil)
+       GlobalEventsJSVM.Init(nil)
        log.Warning("[RPC Backup] --> Initialised JSVM")
 
        resetMainRouter()
```
we get this data race:
```
go test -race ./...
==================
WARNING: DATA RACE
Write at 0x00000253b3f0 by goroutine 360:
  github.com/TykTechnologies/tyk.(*JSVM).Init()
      /Users/test/go/src/github.com/TykTechnologies/tyk/mw_js_plugin.go:322 +0x365
  github.com/TykTechnologies/tyk.setupGlobals()
      /Users/test/go/src/github.com/TykTechnologies/tyk/main.go:186 +0xe48
  github.com/TykTechnologies/tyk.testHttp()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:840 +0xd85
  github.com/TykTechnologies/tyk.TestListener()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:948 +0xf4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous write at 0x00000253b3f0 by goroutine 189:
  github.com/TykTechnologies/tyk.(*JSVM).Init()
      /Users/test/go/src/github.com/TykTechnologies/tyk/mw_js_plugin.go:322 +0x365
  github.com/TykTechnologies/tyk.doReload()
      /Users/test/go/src/github.com/TykTechnologies/tyk/main.go:665 +0x46e
  github.com/TykTechnologies/tyk.reloadLoop()
      /Users/test/go/src/github.com/TykTechnologies/tyk/main.go:704 +0xe2

Goroutine 360 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:192 +0x4d5
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd

Goroutine 189 (running) created at:
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:169 +0x494
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd
==================
==================
WARNING: DATA RACE
Write at 0x00000253b3f8 by goroutine 360:
  github.com/TykTechnologies/tyk.(*JSVM).Init()
      /Users/test/go/src/github.com/TykTechnologies/tyk/mw_js_plugin.go:323 +0x393
  github.com/TykTechnologies/tyk.setupGlobals()
      /Users/test/go/src/github.com/TykTechnologies/tyk/main.go:186 +0xe48
  github.com/TykTechnologies/tyk.testHttp()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:840 +0xd85
  github.com/TykTechnologies/tyk.TestListener()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:948 +0xf4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous write at 0x00000253b3f8 by goroutine 189:
  [failed to restore the stack]

Goroutine 360 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:192 +0x4d5
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd

Goroutine 189 (running) created at:
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:169 +0x494
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd
==================
==================
WARNING: DATA RACE
Write at 0x00000253b400 by goroutine 360:
  github.com/TykTechnologies/tyk.(*JSVM).Init()
      /Users/test/go/src/github.com/TykTechnologies/tyk/mw_js_plugin.go:324 +0x3d5
  github.com/TykTechnologies/tyk.setupGlobals()
      /Users/test/go/src/github.com/TykTechnologies/tyk/main.go:186 +0xe48
  github.com/TykTechnologies/tyk.testHttp()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:840 +0xd85
  github.com/TykTechnologies/tyk.TestListener()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:948 +0xf4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous write at 0x00000253b400 by goroutine 189:
  [failed to restore the stack]

Goroutine 360 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:192 +0x4d5
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd

Goroutine 189 (running) created at:
  github.com/TykTechnologies/tyk.TestMain()
      /Users/test/go/src/github.com/TykTechnologies/tyk/gateway_test.go:169 +0x494
  main.main()
      github.com/TykTechnologies/tyk/_test/_testmain.go:368 +0x1cd
==================
--- FAIL: TestListener (0.56s)
	testing.go:699: race detected during execution of test
FAIL
FAIL	github.com/TykTechnologies/tyk	28.405s
?   	github.com/TykTechnologies/tyk/apidef	[no test files]
ok  	github.com/TykTechnologies/tyk/apidef/importer	1.047s
ok  	github.com/TykTechnologies/tyk/certs	6.588s
ok  	github.com/TykTechnologies/tyk/config	1.071s
?   	github.com/TykTechnologies/tyk/coprocess	[no test files]
ok  	github.com/TykTechnologies/tyk/lint	1.168s
?   	github.com/TykTechnologies/tyk/log	[no test files]
?   	github.com/TykTechnologies/tyk/storage	[no test files]
?   	github.com/TykTechnologies/tyk/user	[no test files]
```